src/tests/test-utils.tsx

```diff
@@ -75,9 +80,56 @@ function customRenderHook<Result, Props>(
   return renderHook(render, { wrapper, ...options })
 }
 
+type MockCallImplementation = {
+  signature: string
+  returnType: string
+  returnValue: unknown
+}
+
+/**
+ * Creates a getWeb3 spy which returns a Web3Provider with a mocked `call` and `resolveName` function.
+ *
+ * @param callImplementations list of supported function calls and the mocked return value. i.e.
+ * ```
+ * [{
+ *   signature: "balanceOf(address)",
+ *   returnType: "uint256",
+ *   returnValue: "200"
+ * }]
+ * ```
+ * @param resolveName mock ens resolveName implementation
+ * @returns web3provider jest spy
+ */
+const mockWeb3Provider = (
```

Comment:
🔥 Can you add a TODO to use this where we've manually done this before? A quick search says it's just the `useSafeTokenAllocation` and `safeCoreSDK` tests.